packages/dashboard/src/LazyDashboard.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { LoadingOverlay } from '@deephaven/components';
+import { Dashboard, DashboardProps } from './Dashboard';
+import { updateDashboardData } from './redux';
+import { DashboardLayoutConfig } from './DashboardLayout';
+
+export interface LazyDashboardProps extends DashboardProps {
+  id: string;
+  isActive: boolean;
+  plugins: JSX.Element[];
+}
+
+export function LazyDashboard({
+  id,
+  isActive,
+  plugins,
+  layoutConfig,
+  ...rest
+}: LazyDashboardProps): JSX.Element {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const dispatch = useDispatch();
+
+  const handleLayoutConfigChange = useCallback(
+    (config?: DashboardLayoutConfig) => {
+      dispatch(updateDashboardData(id, { layoutConfig: config }));
+    },
+    [id, dispatch]
+  );
+
+  if (!isLoaded && isActive) {
+    setIsLoaded(true);
+  }
+
+  if (!isLoaded) {
+    return <LoadingOverlay />;
+  }
```

Comment:
Don't really need separate if blocks for `isLoaded` here, but not a big deal:
```suggestion
  if (!isLoaded) {
    if (isActive) {
      setIsLoaded(true);
    }
    return <LoadingOverlay />;
  }
```

I tend to shy away from calling setState functions from within the render as well, though it does seem to be valid: https://react.dev/reference/react/useState#storing-information-from-previous-renders
My instinct would have been to write this as an effect, e.g.

```
useEffect(() => {
  if (isActive) {
    setIsLoaded(true);
  }
}, [isActive]);
```

Reply:
My instinct is that too, but at some point I read through this article on the new react docs and this is actually in the examples for when not to use an effect.

https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Reply:
Neat